src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -796,7 +800,7 @@ public function isDeprecated()
      */
     public function getDeprecationMessage($id)
     {
-        return str_replace('%service_id%', $id, $this->deprecationTemplate);
+        return null !== $this->deprecationTemplate ? str_replace('%service_id%', $id, $this->deprecationTemplate) : null;
```

Comment:
hum in fact this change is not required and changes the behavior, I'd suggest removing it